### PR TITLE
fix(clearmessage): clear friend did after submit

### PR DESCRIPTION
**Pull request by Jekrimo**
### What this PR does 📖

- clears input form after submit for both add button and input return

### Which issue(s) this PR fixes 🔨

- Resolve #

#304 

### Special notes for reviewers 🗒️

This also fixes a bug that our friend request wasn't being sent on enter

### CODE REVIEW

File: ui/src/components/friends/add.rs [hunk 1]
```diff
@@ -143,6 +143,25 @@ pub fn AddFriend(cx: Scope) -> Element {
...
                     },
+                    reset: request_sent.clone(),
+                    onreturn: move |_| {
+                        match DID::from_str(friend_input.get()) {
+                            Ok(did) => {
+                                if STATIC_ARGS.use_mock {
+                                    let mut ident = Identity::default();
+                                    ident.set_did_key(did);
+                                    state.write().mutate(Action::SendRequest(ident));
+                                    friend_input.set("".to_string());
```
Comment (dariusc93) [inline ui/src/components/friends/add.rs]:
```suggestion
                                    friend_input.set(String::new());
```

File: ui/src/components/friends/add.rs [hunk 2]
```diff
@@ -143,6 +143,25 @@ pub fn AddFriend(cx: Scope) -> Element {
...
+                                } else {
+                                    ch.send(did);
+                                    friend_input.set("".to_string());
```
Comment (dariusc93) [inline ui/src/components/friends/add.rs]:
```suggestion
                                    friend_input.set(String::new());
```

File: ui/src/components/friends/add.rs [hunk 3]
```diff
@@ -159,9 +178,11 @@ pub fn AddFriend(cx: Scope) -> Element {
                                 if STATIC_ARGS.use_mock {
                                     let mut ident = Identity::default();
                                     ident.set_did_key(did);
-                                    state.write().mutate(Action::SendRequest(ident))
+                                    state.write().mutate(Action::SendRequest(ident));
+                                    friend_input.set("".to_string());
```
Comment (dariusc93) [inline ui/src/components/friends/add.rs]:
```suggestion
                                    friend_input.set(String::new());
```

File: ui/src/components/friends/add.rs [hunk 4]
```diff
@@ -159,9 +178,11 @@ pub fn AddFriend(cx: Scope) -> Element {
...
                                 } else {
                                     ch.send(did);
+                                    friend_input.set("".to_string());
```
Comment (dariusc93) [inline ui/src/components/friends/add.rs]:
```suggestion
                                    friend_input.set(String::new());
```

File: ui/src/components/friends/add.rs (hunk 1 above)
Comment (dariusc93) [reply ui/src/components/friends/add.rs]: What might be better would be to clear the input outside of the if statement. Would reduce duplicated code

File: ui/src/components/friends/add.rs (hunk 3 above)
Comment (dariusc93) [reply ui/src/components/friends/add.rs]: ditto

**Review by dariusc93 — COMMENTED** [review]
LGTM. Left a suggestion (though non-blocking) but something i think might be preferable to do would be to use `friend_input.set(String::new());` outside of the if statement so you dont have duplicates. Not blocking either

**Review by dariusc93 — DISMISSED** [review]

**Comment by Jekrimo** [comment]
> LGTM. Left a suggestion (though non-blocking) but something i think might be preferable to do would be to use `friend_input.set(String::new());` outside of the if statement so you dont have duplicates. Not blocking either

Thank you Darius! Got all those fixes in. I knew there had to be a better way!

**Review by lgmarchi — CHANGES_REQUESTED** [review]
I don't know if it is intended, but when I copy own DID and paste, it just clear with return, but it does not clear clicking in the button. 


https://user-images.githubusercontent.com/63157656/220385832-e68dbde2-edc9-4be5-b012-3f107e01fa7f.mov

**Comment by phillsatellite** [comment]
> I don't know if it is intended, but when I copy own DID and paste, it just clear with return, but it does not clear clicking in the button.

Good find @lgmarchi Thanks for testing! 🥲

**Comment by Jekrimo** [comment]
> 

I feel like this is a separate bug for it's own ticket? Or did it function before with an error and such?

**Comment by Jekrimo** [comment]
> I feel like this is a separate bug for it's own ticket? Or did it function before with an error and such? & @lgmarchi

**Comment by lgmarchi** [comment]
> I feel like this is a separate bug for it's own ticket? Or did it function before with an error and such?

I let it more as your decision my friend! If you prefer to solve it, in separated ticket, no problem for me. Just create other ticket for it and comment it if you want and I approve your PR. 

By the way, nice PR man!

**Comment by phillsatellite** [comment]
Seems we might need a ticket for using Enter/Return to send a request which I can write up! On Dev the behavior is 

1. User hits enter/return key nothing happens/no error in console (Same for adding a friend)
2. User clicks "+Add" error will appear in console that says you cannot add yourself. 

Screenshot 2023-02-21 at 1 04 47 PM

**Comment by dariusc93** [comment]
> Seems we might need a ticket for using Enter/Return to send a request which I can write up! On Dev the behavior is
> 
>     1. User hits enter/return key nothing happens/no error in console (Same for adding a friend)
> 
>     2. User clicks "+Add" error will appear in console that says you cannot add yourself.
> 
> 
> Screenshot 2023-02-21 at 1 04 47 PM

Odd. Does that happen when using a different DID?

**Comment by lgmarchi** [comment]
> > Seems we might need a ticket for using Enter/Return to send a request which I can write up! On Dev the behavior is
> > ```
> > 1. User hits enter/return key nothing happens/no error in console (Same for adding a friend)
> > 
> > 2. User clicks "+Add" error will appear in console that says you cannot add yourself.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Screenshot 2023-02-21 at 1 04 47 PM
> 
> Odd. Does that happen when using a different DID?

No Darius, this problem just happen with own DID, but not happens with own DID and when we press return on keyboard.

**Comment by phillsatellite** [comment]
Hey dude! So with the last commits, it did fix the enter key after pasting did issue Lucas found but I did try the same flow but with clicking +Add instead of hitting the enter/return key and the did key persist after clicking 

str 
Copy/paste your own DID
_Click_ +add button
DID key persist 

https://user-images.githubusercontent.com/93608357/220464407-8280182e-d3eb-4916-881c-eca5b67cd539.mov

**Comment by Jekrimo** [comment]
> Hey dude! So with the last commits, it did fix the enter key after pasting did issue Lucas found but I did try the same flow but with clicking +Add instead of hitting the enter/return key and the did key persist after clicking
> 
> str Copy/paste your own DID _Click_ +add button DID key persist

ok awesome, perfect! Back to not being able to add yourself, perfecto

**Review by dariusc93 — DISMISSED** [review]

**Review by lgmarchi — DISMISSED** [review]

**Comment by phillsatellite** [comment]
I might have miscommunicated, when the user copies they're own DID and they _Click_ the "+Add" button the did will persist. After clicking "+add" the did should clear from the input box. Sorry if I confused things at all 

https://user-images.githubusercontent.com/93608357/220669116-ad7a9bbb-1b35-43dd-adf3-d27109f99751.mov

**Review by phillsatellite — APPROVED** [review]

**Comment by phillsatellite** [comment]
Perfect on my side 🌟

**Review by tooshel — APPROVED** [review]

**Review by dariusc93 — APPROVED** [review]

**Pull request merged**